src/Backups/BackupsWorker.cpp

```diff
@@ -563,8 +563,13 @@ void BackupsWorker::writeBackupEntries(BackupMutablePtr backup, BackupEntries &&
             }
         };
 
-        if (always_single_threaded || !backups_thread_pool->trySchedule([job] { job(true); }))
+        if (always_single_threaded)
+        {
             job(false);
+            continue;
+        }
+
+        backups_thread_pool->scheduleOrThrowOnError([job] { job(true); });
```

Comment:
But if we will unable to schedule here the whole backup will fail? Isn't it?

Reply:
This method throws errors only when it cannot allocate a new thread or something similar happened. When pool if full it just blocks and waits for an empty slot.

So yes, if we unable to schedule here the whole backup will fail, but I don't see why previous implementation is any better.